kernel/kernel-authcodeflowproxy-api/src/main/java/io/mosip/kernel/authcodeflowproxy/api/controller/LoginController.java

```diff
@@ -35,10 +35,14 @@
 	private LoginService loginService;
 
 	@GetMapping(value = "/login/{redirectURI}")
-	public void login(@CookieValue("state") String state, @PathVariable("redirectURI") String redirectURI,
+	public void login(@CookieValue("state") String state, @PathVariable("redirectURI") String redirectURI, @RequestParam(name = "state",required = false) String stateParam,
 			HttpServletResponse res) throws IOException {
-		String uri = loginService.login(redirectURI, state);
-
+		String stateValue = state==null?stateParam:state;
+		String uri = loginService.login(redirectURI, stateValue);
```

Comment:
can we chk if the stateValue is uuid? If a state comes with new line character then it will result in a vulnerability

Reply:
done